static/app/components/group/tagFacets/tagFacetsDistributionMeter.tsx

```diff
@@ -0,0 +1,267 @@
+                    aria-label={t('Other')}
+                    color={colors[colors.length - 1]}
+                  />
+                ) : (
+                  <Segment
+                    aria-label={t(
+                      'Add the %s %s segment tag to the search query',
+                      title,
+                      value.value
+                    )}
+                    color={colors[index]}
+                    {...segmentProps}
+                  >
+                    {index === 0 ? `${pctLabel}%` : null}
+                  </Segment>
+                )}
+              </Tooltip>
+            </div>
+          );
+        })}
+      </SegmentBar>
+    );
+  }
+
+  function renderLegend(tooltip: boolean = false) {
+    return (
+      <LegendGrid>
+        {segments.map((segment, index) => {
+          const pctLabel = Math.floor(percent(segment.count, totalValues));
+          return (
+            <LegendRow key={`segment-${segment.name}-${index}`} tooltip={tooltip}>
+              <LegendDot color={colors[index]} />
+              <LegendText>{segment.name}</LegendText>
+              <LegendPercent>{`${pctLabel}%`}</LegendPercent>
+            </LegendRow>
+          );
+        })}
+      </LegendGrid>
+    );
+  }
+
+  const totalVisible = segments.reduce((sum, value) => sum + value.count, 0);
+  const hasOther = totalVisible < totalValues;
+
+  if (hasOther) {
+    segments.push({
+      isOther: true,
+      name: t('Other'),
+      value: 'other',
+      count: totalValues - totalVisible,
+      url: '',
+    });
+  }
+
+  return (
+    <TagSummary>
+      {renderTitle()}
+      {!expanded ? renderSegments() : null}
+      {expanded ? renderLegend() : null}
+    </TagSummary>
+  );
+}
+
+export default TagFacetsDistributionMeter;
+
+const TagSummary = styled('div')`
+  margin-bottom: ${space(2)};
+`;
+
+const SegmentBar = styled('div')`
+  display: flex;
+  overflow: hidden;
+`;
+
+const Title = styled('div')`
+  display: flex;
+  font-size: ${p => p.theme.fontSizeSmall};
+  justify-content: space-between;
+  margin-bottom: ${space(0.25)};
+  line-height: 1.1;
+`;
+
+const TitleType = styled('div')`
+  color: ${p => p.theme.textColor};
+  font-weight: bold;
+  ${p => p.theme.overflowEllipsis};
+  font-size: ${p => p.theme.fontSizeSmall};
+`;
+
+const TitleDescription = styled('div')`
+  display: flex;
+  color: ${p => p.theme.gray300};
+  text-align: right;
+  font-size: ${p => p.theme.fontSizeSmall};
+`;
+
+const Label = styled('div')`
+  ${p => p.theme.overflowEllipsis};
+  max-width: 150px;
+`;
+
+const OtherSegment = styled('span')<{color: string}>`
+  display: block;
+  width: 100%;
+  height: 16px;
+  color: inherit;
+  outline: none;
+  background-color: ${p => p.color};
+`;
+
+const Segment = styled(Link, {shouldForwardProp: isPropValid})<{color: string}>`
+  &:hover {
+    color: ${p => p.theme.white};
+  }
+  display: block;
+  width: 100%;
+  height: 16px;
+  color: ${p => p.theme.white};
```

Comment:
If the colour is already white, then I don't think we need the hover styling

Reply:
There is a hover color on the base component which makes it blue. I had to add this to prevent it from turning blue on hover 😄